### ISSUE TITLE
[BUG] Locations weren't updating anymore

### ISSUE THREAD

**Issue by mrdanleey** [Myzel394/locus#137]
## OS

* [x] Android
* [ ] iOS

## Android specific
If you ticked `Android`, please specify on which Android flavor this issue happens

* [x] Samsung Android (Samsung Phones)
* [ ] MIUI (Xiaomi Phones)
* [ ] Stock ROM (Google Phones)
* [ ] Huawei Android (Huawei Phones)
* [ ] GrapheneOS
  * [ ] I do have Google Play Services installed
* [x] LineageOS

## Describe the bug
This weekend I discovered that my locations on all 3 phone's did not update anymore.
I did not get any updates only the error that there was no data and that is was possible that the tasks were not running anymore.

So I did some research and discoverd that the relayservers I selected in the app were not working anymore. I unselected the relausevers and selected new ones. The ones that are on top of the list. I then generatied a new link and added it to the other phones and visa versa. That made it working againg.  

This made me think about how the app is working. Is looks like the app generates the link with the selected relayservers included. Because the locationsharing dit not work by only changing the relayservers in the app and the share. I had to generate a new link to make the share work again. Is this how it works? And what if my wife is sharing her location and it stops sharing because the relayservers selected on her app aren't working anymore? She had to do several things to make it work again. That's not handy but maybe the only way? Or is this a bug? Or is this how it works but can we figure out another way to fix the sharing when a relayserver stops working. Ofcourse it was strange that al 5 relayservers did not work anymore. But it made me wonder what was happening.

## To Reproduce
1. Just wait thill the relaysevers are not working anymore.

## Expected behavior
I hope that is it possible to select new relayservers in the app and the share to make it work again without generating a new link, or maybe that the link is automatically changing for the phones who are follwing the task. 
## Priority
Please set a realistic priority on how important this issue is to you. We will fix all bugs, but urgent bugs (such as the app not starting at all for example) are more important than fixing some styles for example.

* [ ] Low
* [x] Medium
* [ ] High
* [ ] Urgent
 
## App Version
0.15.2 both versions

**Comment by Myzel394** [comment]
> This made me think about how the app is working. Is looks like the app generates the link with the selected relayservers included.

You are completely correct, the relay servers are used for both sharing the link and the actual location data. To share the your Share without relying on a relay server, you'll need to share the view file.

> I hope that is it possible to select new relayservers in the app and the share to make it work again without generating a new link, or maybe that the link is automatically changing for the phones who are follwing the task.

You will definitely need to regenerate the link and also share the view file again if you did that. The link and the share both contain the selected relays in them, to notify the receiver which relays to get your location from.

Relay servers going down is a serious problem I'm facing with Locus. I always advise to use the top most relays as they seem to be the most reliable. Otherwise, for tech-savy people, a different solution can be to self host a nostr relays.

However, I'll need to find a different way. I don't have too much time now but I hope I will be able to resolve this issue this year.
I'm sorry to not be able to provide a quick fix, but maybe my suggested solutions will work for you :)
